File: frontend/lib/models/agenda/service_scheduling_mode.dart
/// Cómo se agenda un servicio en reservas públicas.
enum ServiceSchedulingMode {
  general,
  byStaff;

  static ServiceSchedulingMode fromApi(String? value) {
    if (value == null) return ServiceSchedulingMode.general;
    switch (value.toUpperCase()) {
      case 'BY_STAFF':
        return ServiceSchedulingMode.byStaff;
      default:
        return ServiceSchedulingMode.general;
    }
  }

  String toApi() => this == ServiceSchedulingMode.byStaff ? 'BY_STAFF' : 'GENERAL';
}

File: backend/src/main/java/com/botai/domain/agenda/model/ServiceSchedulingMode.java
package com.botai.domain.agenda.model;

public enum ServiceSchedulingMode {
    GENERAL,
    BY_STAFF;

    public static ServiceSchedulingMode fromString(String value) {
        if (value == null || value.isBlank()) {
            return GENERAL;
        }
        return ServiceSchedulingMode.valueOf(value.trim().toUpperCase());
    }
}
